cgen: error: '{' expected (got ";")

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.9 496451e, press to see full `v doctor` output</summary>

|V full version      |V 0.4.9 4fcd94a.496451e
|:-------------------|:-------------------
|OS                  |windows, Њ ©Єа®б®дв Windows 10 Pro 19045 64-а §ап¤­ п
|Processor           |28 cpus, 64bit, little endian, Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz  
|Memory              |16.35GB/31.86GB
|                    |
|V executable        |C:\Dev\Progs\v\v.exe
|V last modified time|2025-01-17 20:10:01
|                    |
|V home dir          |OK, value: C:\Dev\Progs\v
|VMODULES            |OK, value: C:\Users\koplenov\.vmodules
|VTMP                |OK, value: C:\Users\koplenov\AppData\Local\Temp\v_0
|Current working dir |OK, value: C:\Dev\Zone\mam_v\treevee\src\ast\repro2
|                    |
|Git version         |git version 2.42.0.windows.2
|V git status        |weekly.2025.1-72-g496451ec
|.git/config present |true
|                    |
|cc version          |N/A
|gcc version         |gcc (MinGW-W64 x86_64-ucrt-posix-seh, built by Brecht Sanders, r4) 13.2.0
|clang version       |clang version 18.1.2
|msvc version        |N/A
|tcc version         |tcc version 0.9.27 (x86_64 Windows)
|tcc git status      |thirdparty-windows-amd64 b425ac82
|emcc version        |N/A
|glibc version       |N/A

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  .\main.v && C:\Dev\Zone\mam_v\treevee\src\ast\repro2\main.exe`
```v
module main

pub struct Tree {}
pub fn (tree &Tree) str() string {
	return ""
}

pub struct TreeTwo {
	Tree
}
pub fn TreeTwo.from_string(tree_string string) !&TreeTwo{
	return &TreeTwo{}
}

fn main(){
	tree := TreeTwo.from_string("") or { panic(err) }
	println(tree.str())
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: C:/Users/koplenov/AppData/Local/Temp/v_0/main.01JHTY7RB3A0Y45GT5G6B5CVZ2.tmp.c:1727: warning: implicit declaration of function 'string_repeat'
cc: C:/Users/koplenov/AppData/Local/Temp/v_0/main.01JHTY7RB3A0Y45GT5G6B5CVZ2.tmp.c:1727: error: '{' expected (got ";")
... (the original output was 3 lines long, and was truncated to 2 lines)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

Successful compilation


But this compiles:
```v
module main

pub struct Tree {}
pub fn (tree &Tree) my_str() string {
	return ""
}

pub struct TreeTwo {
	Tree
}
pub fn TreeTwo.from_string(tree_string string) !&TreeTwo{
	return &TreeTwo{}
}

fn main(){
	tree := TreeTwo.from_string("") or { panic(err) }
	println(tree.my_str())
}

```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1424,7 +1424,8 @@ fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) ast.
 		c.builtin_args(mut node, fn_name, func)
 		if c.pref.skip_unused && !c.is_builtin_mod && c.mod != 'math.bits'
 			&& node.args[0].expr !is ast.StringLiteral {
-			if (node.args[0].expr is ast.CallExpr && node.args[0].expr.name == 'str')
+			if (node.args[0].expr is ast.CallExpr && node.args[0].expr.is_method
+				&& node.args[0].expr.name == 'str')
 				|| !c.table.sym(c.unwrap_generic(node.args[0].typ)).has_method('str') {
 				c.table.used_features.auto_str = true
 			} else {

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1424,7 +1424,8 @@ fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) ast.
 		c.builtin_args(mut node, fn_name, func)
 		if c.pref.skip_unused && !c.is_builtin_mod && c.mod != 'math.bits'
 			&& node.args[0].expr !is ast.StringLiteral {
-			if !c.table.sym(c.unwrap_generic(node.args[0].typ)).has_method('str') {
+			if (node.args[0].expr is ast.CallExpr && node.args[0].expr.name == 'str')
+				|| !c.table.sym(c.unwrap_generic(node.args[0].typ)).has_method('str') {
 				c.table.used_features.auto_str = true
 			} else {
 				if node.args[0].typ.has_option_or_result() {

--- a/vlib/v/tests/skip_unused/auto_str.run.out
+++ b/vlib/v/tests/skip_unused/auto_str.run.out
@@ -1,0 +1,3 @@
+&TreeTwo{
+    Tree: 
+}

--- a/vlib/v/tests/skip_unused/auto_str.skip_unused.run.out
+++ b/vlib/v/tests/skip_unused/auto_str.skip_unused.run.out
@@ -1,0 +1,3 @@
+&TreeTwo{
+    Tree: 
+}

--- a/vlib/v/tests/skip_unused/auto_str.vv
+++ b/vlib/v/tests/skip_unused/auto_str.vv
@@ -1,0 +1,20 @@
+module main
+
+pub struct Tree {}
+
+pub fn (tree &Tree) str() string {
+	return ''
+}
+
+pub struct TreeTwo {
+	Tree
+}
+
+pub fn TreeTwo.from_string(tree_string string) !&TreeTwo {
+	return &TreeTwo{}
+}
+
+fn main() {
+	tree := TreeTwo.from_string('') or { panic(err) }
+	println(tree.str())
+}

